pkg/controllers/workgenerator/controller_integration_test.go

```diff
@@ -68,45 +70,141 @@ var _ = Describe("Test clusterSchedulingPolicySnapshot Controller", func() {
+				return k8sClient.Get(ctx, types.NamespacedName{Name: fmt.Sprintf(fleetv1beta1.FirstWorkNameFmt, testCRPName), Namespace: namespaceName}, &work)
+			}, timeout, interval).Should(Succeed(), "Failed to get the expected work in hub cluster")
+			By(fmt.Sprintf("work %s is created in %s", work.Name, work.Namespace))
+			Eventually(func() error {
+				return k8sClient.Get(ctx, types.NamespacedName{Name: fmt.Sprintf(fleetv1beta1.WorkNameWithSubindexFmt, testCRPName, 1), Namespace: namespaceName}, &work)
+			}, timeout, interval).Should(Succeed(), "Failed to get the expected work in hub cluster")
+			By(fmt.Sprintf("work %s is created in %s", work.Name, work.Namespace))
 		})
 
-		It("Should not handle the binding with state scheduled", func() {
-			// create master resource snapshot with 1 number of resources
+		It("Should handle the case that the snapshot is deleted", func() {
+			// generate master resource snapshot name but not create it
 			masterSnapshot := generateResourceSnapshot(1, 1, 0, [][]byte{
 				testClonesetCRD, testNameSpace, testCloneset,
 			})
-			Expect(k8sClient.Create(ctx, masterSnapshot)).Should(Succeed())
-			By(fmt.Sprintf("master resource snapshot  %s created", masterSnapshot.Name))
-			// create a scheduled binding
-			binding = generateClusterResourceBinding(fleetv1beta1.BindingStateScheduled, masterSnapshot.Name, memberClusterName)
+			// create a scheduled binding pointing to the non exist master resource snapshot
+			binding = generateClusterResourceBinding(fleetv1beta1.BindingStateBound, masterSnapshot.Name, memberClusterName)
 			Expect(k8sClient.Create(ctx, binding)).Should(Succeed())
 			By(fmt.Sprintf("resource binding  %s created", binding.Name))
-			// check the work is not created since the binding state is not bound
+			// check the work is not created since there is no resource snapshot
 			work := v1alpha1.Work{}
 			Consistently(func() bool {
 				err := k8sClient.Get(ctx, types.NamespacedName{Name: fmt.Sprintf(fleetv1beta1.FirstWorkNameFmt, testCRPName), Namespace: namespaceName}, &work)
 				return apierrors.IsNotFound(err)
 			}, duration, interval).Should(BeTrue(), "controller should not create work in hub cluster until all resources are created")
-			// binding should not have any finalizers
+			// check the binding status
+			var diff string
+			wantMC := fleetv1beta1.ResourceBindingStatus{
+				Conditions: []metav1.Condition{
+					{
+						Type:               string(fleetv1beta1.ResourceBindingBound),
+						Status:             metav1.ConditionFalse,
+						Reason:             syncWorkFailed,
+						ObservedGeneration: binding.GetGeneration(),
+					},
+				},
+			}
 			Expect(k8sClient.Get(ctx, types.NamespacedName{Name: binding.Name}, binding)).Should(Succeed())
-			Expect(len(binding.Finalizers)).Should(Equal(0))
+			diff = cmp.Diff(wantMC, binding.Status, ignoreConditionOption)
+			Expect(diff).Should(BeEmpty(), fmt.Sprintf("binding(%s) mismatch (-want +got):\n%s", binding.Name, diff))
```

Comment:
Could change to

Expect(k8sClient.Get(ctx, types.NamespacedName{Name: binding.Name}, binding)).Should(Succeed())
wantMC := fleetv1beta1.ResourceBindingStatus{...}
diff := cmp.Diff(wantMC, binding.Status, ignoreConditionOption)

move the wantMC closer to where it's used

Reply:
the diff line used both `binding` and `wantMC`. Is there any reason why wantMC should be the closest to the cmp instead of binding? Is that it's more complicated?